File: app/src/main/java/ca/rmen/android/poetassistant/widget/CABTextView.java
package ca.rmen.android.poetassistant.widget;

import android.content.Context;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;

/**
 * https://code.google.com/p/android/issues/detail?id=23381
 */
public class CABTextView extends AppCompatTextView implements HackFor23381 {
    private boolean shouldWindowFocusWait;

    public CABTextView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public CABTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CABTextView(Context context) {
        super(context);
    }

    @Override
    public void setWindowFocusWait(boolean shouldWindowFocusWait) {
        this.shouldWindowFocusWait = shouldWindowFocusWait;
    }

    @Override
    public void onWindowFocusChanged(boolean hasWindowFocus) {
        if(!shouldWindowFocusWait) {
            super.onWindowFocusChanged(hasWindowFocus);
        }
    }
}

File: app/src/main/kotlin/ca/rmen/android/poetassistant/widget/CABTextView.kt
package ca.rmen.android.poetassistant.widget

import android.content.Context
import android.support.v7.widget.AppCompatTextView
import android.util.AttributeSet

/**
 * https://code.google.com/p/android/issues/detail?id=23381
 */
class CABTextView @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0
) : AppCompatTextView(context, attrs, defStyleAttr), HackFor23381 {
    private var shouldWindowFocusWait: Boolean = false

    override fun setWindowFocusWait(shouldWindowFocusWait: Boolean) {
        this.shouldWindowFocusWait = shouldWindowFocusWait
    }

    override fun onWindowFocusChanged(hasWindowFocus: Boolean) {
        if (!shouldWindowFocusWait) {
            super.onWindowFocusChanged(hasWindowFocus)
        }
    }
}
